src/cart/crt_hg.c

```diff
@@ -1189,6 +1192,22 @@ crt_hg_req_send(struct crt_rpc_priv *rpc_priv)
 		RPC_ERROR(rpc_priv,
 			  "HG_Forward failed, hg_ret: %d\n",
 			  hg_ret);
+		if (hg_ret == HG_FI_EAGAIN) {
+			/*Put in retry queue.*/
+			req = &rpc_priv->crp_pub;
+			ctx = req->cr_ctx;
+			D_MUTEX_LOCK(&ctx->cc_mutex);
+			D_INIT_LIST_HEAD(&rpc_priv->crp_retry);
+			d_list_add_tail(&rpc_priv->crp_retry, &ctx->cc_retry);
+			D_MUTEX_UNLOCK(&ctx->cc_mutex);
+			RPC_ADDREF(rpc_priv);
+			RPC_TRACE(DB_TRACE, rpc_priv,
+			  	"will retry send to rank later%d uri: %s\n",
```

Comment:
(style)  code indent should use tabs where possible
(style)  please, no space before tabs